test/chat_api/notes_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule ChatApi.NotesTest do
+  use ChatApi.DataCase
+
+  import ChatApi.Factory
+  alias ChatApi.Notes
+
+  describe "notes" do
+    alias ChatApi.Notes.Note
+
+    @update_attrs %{body: "some updated body"}
+    @invalid_attrs %{body: nil}
+
+    setup do
+      account = insert(:account)
+      agent = insert(:user, account: account)
+      customer = insert(:customer, account: account)
+      note = insert(:note, body: "note_body", account: account, author: agent, customer: customer)
+
+      {:ok, agent: agent, account: account, customer: customer, note: note}
+    end
+
+    test "list_notes/1 returns all notes for the given account and customer", %{
+      note: note,
+      account: account,
+      customer: customer
+    } do
+      note_ids =
+        Notes.list_notes_for_customer(%{account_id: account.id, customer_id: customer.id})
+        |> Enum.map(& &1.id)
+
+      assert note_ids == [note.id]
+    end
+
+    test "get_note!/1 returns the note with given id", %{note: note} do
+      found = Notes.get_note!(note.id)
+      assert found.id == note.id
+      assert found.body == note.body
+    end
+
+    test "create_note/1 with valid data creates a note", %{
+      account: account,
+      customer: customer,
+      agent: agent
+    } do
+      assert {:ok, %Note{} = note} =
+               Notes.create_note(%{
+                 body: "some body",
+                 account_id: account.id,
+                 customer_id: customer.id,
+                 author_id: agent.id
+               })
+
+      assert note.body == "some body"
+      assert note.customer_id == customer.id
+      assert note.author_id == agent.id
+      assert note.account_id == account.id
+    end
+
+    test "create_note/1 with invalid data returns error changeset" do
```

Comment:
can we also test that attempting to create a note with an empty string body also returns an error changeset?